src/AppNavigator.tsx

```diff
@@ -2,82 +2,99 @@ import {useReduxDevToolsExtension} from '@react-navigation/devtools'
 import {NavigationContainer, useNavigationContainerRef} from '@react-navigation/native'
 import {createStackNavigator} from '@react-navigation/stack'
 import React, {useEffect} from 'react'
-import type {IntlShape} from 'react-intl'
 import {defineMessages, useIntl} from 'react-intl'
-import {Alert, AppState} from 'react-native'
-import {useQueryClient} from 'react-query'
-import {useDispatch, useSelector} from 'react-redux'
+import {Alert, AppState, AppStateStatus, Platform} from 'react-native'
+import RNBootSplash from 'react-native-bootsplash'
+import {useSelector} from 'react-redux'
 
-import {CreatePinScreen, PinLoginScreen} from './auth'
-import {useBackgroundTimeout} from './auth'
+import {PinLoginScreen, useAuthWithOs, useBackgroundTimeout, useCanEnableAuthOs} from './auth'
 import {useAuth} from './auth/AuthProvider'
-import {BiometricAuthScreen} from './BiometricAuth'
+import {EnableLoginWithPin} from './auth/EnableLoginWithPin'
 import {FirstRunNavigator} from './FirstRun/FirstRunNavigator'
-import {useHasWallets} from './hooks'
-import {errorMessages} from './i18n/global-messages'
-import {checkBiometricStatus, reloadAppSettings, setSystemAuth, showErrorDialog} from './legacy/actions'
+import {useAuthMethod} from './hooks'
+import globalMessages from './i18n/global-messages'
 import {DeveloperScreen} from './legacy/DeveloperScreen'
-import {canBiometricEncryptionBeEnabled, recreateAppSignInKeys} from './legacy/deviceSettings'
 import env from './legacy/env'
-import KeyStore from './legacy/KeyStore'
-import {
-  canEnableBiometricSelector,
-  installationIdSelector,
-  isAppSetupCompleteSelector,
-  isMaintenanceSelector,
-  isSystemAuthEnabledSelector,
-} from './legacy/selectors'
+import {installationIdSelector, isMaintenanceSelector, isTosAcceptedSelector} from './legacy/selectors'
 import {isEmptyString} from './legacy/utils'
 import MaintenanceScreen from './MaintenanceScreen'
 import {AppRoutes} from './navigation'
+import {OsLoginScreen} from './OsAuth'
+import {useStorage} from './Storage'
 import StorybookScreen from './StorybookScreen'
 import {WalletInitNavigator} from './WalletInit/WalletInitNavigator'
 import {WalletNavigator} from './WalletNavigator'
-import {walletManager} from './yoroi-wallets'
 
 const IS_STORYBOOK = env.getBoolean('IS_STORYBOOK', false)
 
 export const AppNavigator = () => {
+  const strings = useStrings()
+  const storage = useStorage()
+  const [isReady, setIsReady] = React.useState(false)
   useAutoLogout()
+  useHideScreenInAppSwitcher()
+  const {authMethod} = useAuthMethod(storage)
```

Comment:
already suspense, remove curly braces here and from hook

Reply:
same thing need to force refetch in the application settings when focusing back